geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANPropagationLoopBackDUnitTest.java

```diff
@@ -433,6 +455,259 @@ public void unstartedSenderShouldNotAddReceivedEventsIntoTmpDropped() throws Exc
+   * Start the sender, and make sure the dropped events have been removed and that
+   * those events are not sent to LN.
+   */
+  // TODO Should we keep this test case?
+  @Test
+  public void stoppedSenderShouldAddReceivedEventsIntoTmpDropped()
+      throws Exception {
+    Integer lnPort = (Integer) vm0.invoke(() -> WANTestBase.createFirstLocatorWithDSId(2));
+    Integer nyPort = (Integer) vm1.invoke(() -> WANTestBase.createFirstRemoteLocator(1, lnPort));
+
+    // create receiver on site-ln and site-ny
+    createCacheInVMs(lnPort, vm2, vm4);
+    createReceiverInVMs(vm2, vm4);
+    createCacheInVMs(nyPort, vm3, vm5);
+    createReceiverInVMs(vm3, vm5);
+
+    // create senders on site-ln, Note: sender-id is its destination, i.e. ny
+    vm2.invoke(() -> WANTestBase.createSender("ny", 1, true, 100, 10, false, false, null, true));
+    vm4.invoke(() -> WANTestBase.createSender("ny", 1, true, 100, 10, false, false, null, true));
+
+    // create senders on site-ny, Note: sender-id is its destination, i.e. ln but configure to not
+    // queue dropped events
+    vm3.invoke(() -> WANTestBase.createSender("ln", 2, true, 100, 10, false, false, null, true));
+    vm5.invoke(() -> WANTestBase.createSender("ln", 2, true, 100, 10, false, false, null, true));
+
+    // create PR on site-ln
+    vm2.invoke(() -> WANTestBase.createPartitionedRegion(getTestMethodName() + "_PR", "ny", 1, 100,
+        isOffHeap()));
+    vm4.invoke(() -> WANTestBase.createPartitionedRegion(getTestMethodName() + "_PR", "ny", 1, 100,
+        isOffHeap()));
+
+    // create PR on site-ny
+    vm3.invoke(() -> WANTestBase.createPartitionedRegion(getTestMethodName() + "_PR", "ln", 1, 100,
+        isOffHeap()));
+    vm5.invoke(() -> WANTestBase.createPartitionedRegion(getTestMethodName() + "_PR", "ln", 1, 100,
+        isOffHeap()));
+
+    // start sender on site-ln
+
+    startSenderInVMs("ny", vm2, vm4);
+
+    // start sender on site-ny
+    startSenderInVMsAsync("ln", vm3, vm5);
+
+    // stop sender on site-ny
+    stopSenderInVMsAsync("ln", vm3, vm5);
+
+    // do next 100 puts on site-ny
+    vm3.invoke(() -> WANTestBase.doPutsFrom(getTestMethodName() + "_PR", 0, 100));
+
+    // verify site-ny has 100 entries
+    vm3.invoke(() -> WANTestBase.validateRegionSize(getTestMethodName() + "_PR", 100));
+    vm5.invoke(() -> WANTestBase.validateRegionSize(getTestMethodName() + "_PR", 100));
+
+    // start sender on site-ny
+    startSenderInVMsAsync("ln", vm3, vm5);
+
+    // verify site-ln has not received the events from site-ny because they were dropped
+    vm2.invoke(() -> WANTestBase.validateRegionSize(getTestMethodName() + "_PR", 0));
+    vm4.invoke(() -> WANTestBase.validateRegionSize(getTestMethodName() + "_PR", 0));
+
+    // verify tmpDroppedEvents is 0 now at site-ny
+    vm3.invoke(() -> WANTestBase.verifyTmpDroppedEventSize("ln", 0));
+    vm5.invoke(() -> WANTestBase.verifyTmpDroppedEventSize("ln", 0));
+
   }
```

Comment:
You need to add a test case: 
1) stop sender, after 15 seconds, do a few put(), then retart the sender and make sure these events won't stay in secondary queue for ever.

Reply:
That is tested in test case restartedSenderShouldAddReceivedEventsIntoTmpDropped after the 15 seconds sleep.